### PR TITLE
Elsi restruct

### PR DESCRIPTION
**Pull request by aradi**
The refactoring is not perfect, but should be OK for the moment. I hope, I did not break anything.

### CODE REVIEW

File: prog/dftb+/include/common.fypp [hunk 1]
```diff
@@ -77,5 +77,24 @@
...
+#! Creates a stub subroutine with given arguments which calls raises an error.
+#!
+#! Args:
+#!     NAME: Name of the stub subroutine to create.
+#!     ARGTYPES: List of subroutine argument types.
+#!
+#:def STUB_SUBROUTINE(NAME, ARGTYPES)
+#:set args = ["arg{}".format(iarg + 1) for iarg in range(len(ARGTYPES))]
+subroutine ${NAME}$(${", ".join(args)}$)
+    use message, only : error
```
Comment (bhourahine) [inline prog/dftb+/include/common.fypp]: This doesn't work with the current dependency generation mechanism from cr_makedep as the F90 routine using the STUB macro won't have a use at the top for message

Comment (aradi) [reply prog/dftb+/include/common.fypp]: You're right, I'll fix it.

**Comment by aradi** [comment]
Now, it should be fine :wink:

**Comment by aradi** [comment]
Sorry, not yet, I am fixing...

**Comment by aradi** [comment]
OK, now the ELSI interface is also OK. I changed to exact argument names (instead of generated ones) to enable for named arguments in ELSI-calls (although not done it the code yet).

**Pull request closed without merging**